### PR TITLE
Docs/igw docs day refactor

### PR DESCRIPTION
**Pull request by trujillo-adam**
### Description

This PR refactors the existing ingress gateway documentation into overview, usage, and configuration reference content types. We edited some content for clarity and so that it fit into the content type paradigm, but it is mostly unchanged. Please review for accuracy.  

Preview pages:

- [Overview](https://consul-eztxi0f5c-hashicorp.vercel.app/consul/docs/connect/gateways/ingress-gateway)
- [Main usage page](https://consul-eztxi0f5c-hashicorp.vercel.app/consul/docs/connect/gateways/ingress-gateway/ingress-gateways-usage)
- [Secondary usage page](https://consul-eztxi0f5c-hashicorp.vercel.app/consul/docs/connect/gateways/ingress-gateway/ingress-gateways-tls-external-service): For this page, please verify the order of the steps. It follows the order of the existing page, but a few questions emerged as we created the draft. Additionally, please verify the description under the 'Configure static SDS clusters' step. 
- [Configuration reference](https://consul-eztxi0f5c-hashicorp.vercel.app/consul/docs/connect/config-entries/ingress-gateway)

### PR Checklist

* [ ] updated test coverage
* [X] external facing docs updated
* [X] appropriate backport labels added
* [X] not a security concern

### CODE REVIEW

File: website/content/docs/connect/config-entries/ingress-gateway.mdx
Comment (missylbytes) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:254]: This should be a `:` not a `=`

Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:254]:
```suggestion
  namespace: "<namespace to register the gateway>" 
```

File: website/content/api-docs/agent/service.mdx
Comment (nathancoleman) [inline website/content/api-docs/agent/service.mdx:638]:
```suggestion
  - `"ingress-gateway"` for instances of an [ingress gateway](/consul/docs/connect/gateways/ingress-gateway)
```

File: website/data/docs-nav-data.json
Comment (nathancoleman) [inline website/data/docs-nav-data.json:689]: The paths here feel redundant to me. Would it make sense to have the following?
- connect/gateways/ingress-gateway/usage
- connect/gateways/ingress-gateway/tls-external-service

File: website/content/docs/connect/gateways/ingress-gateway/ingress-gateways-usage.mdx
Comment (nathancoleman) [inline website/content/docs/connect/gateways/ingress-gateway/ingress-gateways-usage.mdx:101]: I'm imagining this isn't meant to be here

```suggestion
- `-register`
```

**Review by nathancoleman — CHANGES_REQUESTED** [review]

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (im2nguyen) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:108]:
```suggestion
$ consul connect envoy -gateway=ingress -register \
   -service ingress-service \
   -address '{{ GetInterfaceIP "eth0" }}:8888'
```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (im2nguyen) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:93]:
```suggestion
$ consul connect envoy -gateway=ingress ingress-service
```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (im2nguyen) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:73]:
```suggestion
- [`Kind`](/consul/docs/services/configuration/services-configuration-reference#kind): The field must be set to `ingress-gateway`.
- [`Name`](/consul/docs/services/configuration/services-configuration-reference#name): The name should match the value specified for the `Name` field in the configuration entry.
```

**Review by im2nguyen — APPROVED** [review]
Looks good, a couple small nits, but pre-approving so you're not blocked from merging

I noticed you used `<CodeBlock>` instead of `<CodeBlockConfig>`, can you please change this before merging? Also, this is a nit, but newlines are inconsistent after headings and codefences, would be nice if we can be consistent with a single new line after each one

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:52]:
```suggestion
<CodeBlockConfig heading="public-ingress-service.hcl">
```

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:115]:
```suggestion
</CodeBlockConfig>
```

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (im2nguyen) [reply website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:52]: there's a couple of these, so that's why i didn't do suggestions for all of them 😛

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:136]:
```suggestion
<CodeBlockConfig heading="certs/sds-auth-cert.json">
```

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:157]:
```suggestion
</CodeBlockConfig>

```

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:160]:
```suggestion
<CodeBlockConfig heading="/certs/sds-validation.json">
```

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:178]:
```suggestion
</CodeBlockConfig>
```

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:211]:
```suggestion
<CodeBlockConfig heading="public-ingress-cfg.hcl">
```

File: website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx:233]:
```suggestion
</CodeBlockConfig>
```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:23]:
```suggestion
## Requirements

```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:39]:
```suggestion

### Define an ingress gateway configuration entry

```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:52]:
```suggestion

###  Register an ingress gateway configuration entry
```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:61]:
```suggestion

## Deploy an ingress gateway service
```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:66]:
```suggestion

### Create a service definition for the ingress gateway 
```

File: website/content/docs/connect/gateways/ingress-gateway/usage.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/ingress-gateway/usage.mdx:86]:
```suggestion

## Start an Envoy proxy
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:468]:
```suggestion

#### Values
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:646]:
```suggestion

#### Values
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:650]:
```suggestion

### `Listeners[].Protocol`
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:952]:
```suggestion

### `kind`
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1030]:
```suggestion

### `spec.tls.enabled`

Enables and disables TLS for the configuration entry. Set to `true` to enable built-in TLS for every listener on the gateway. TLS is disabled by default.

When enabled, Consul adds each host defined in every service's `Hosts` field to the gateway's x509 certificate as a DNS subject alternative name (SAN). 

#### Values

 - Default: `false`
 - Data type: boolean 

### `spec.tls.tlsMinVersion`
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1048]:
```suggestion

### `spec.tls.tlsMaxVersion`

Specifies the maximum TLS version supported for gateway listeners.

#### Values
```

**Pull request merged**
